File: Swift-C++-Testsuite/test/error-handling/Swift/ThrowCustomError.swift
@main
struct MyMain {
    enum VendingMachineError: Error {
        case invalidSelection
        case insufficientFunds(coinsNeeded: Int)
        case outOfStock
    }
    static func main() {
        do {
            try vendingMachineInteraction()
        } catch {
            let e = error
        }

    }

    static func vendingMachineInteraction() throws {
        throw VendingMachineError.insufficientFunds(coinsNeeded: 5)
    }
}

File: AutoTranspiledTests/Swift-to-Cpp/try-0/ThrowCustomError.cpp

#include <iostream>

enum VendingMachineError {
  InvalidSelection,
  InsufficientFunds,
  OutOfStock
};

int main() {
  try {
    throw VendingMachineError::InsufficientFunds;
  } catch (const VendingMachineError& e) {
    std::cout << e;
  }
  return 0;
}